File: R/012.R
NUM <- 500

create_triangle_number <- function(n){
  if (n >= 1) {
    return(sum(1:n))  
  }
  return(NULL)
}

i <- 1
divisors <- 0

while (divisors < NUM) {
  n <- create_triangle_number(i)
  divisors <- 0
  
  for (j in 1:sqrt(n)) {
    if (n %% j == 0) {
      divisors <- divisors + 2
    }
  }
  i <- i + 1
}

print(paste0('triangle number: ', n, ', and # of divisors is ', divisors))

File: Python/012.py
import math
a = 0
s = 0
d = 0
i = 1
c = 0
while i > 0:
    a += i
    c = 0

    for j in range(1, int(math.sqrt(a) + 1)):
        if a % j == 0:
            c += 1
    if d < c:
        s = (c - 1) * 2 + 1
        d = c
        print(a, s)
    if s > 500:
        print(a, s)
        break
# if a % 500 == 0:
# print(a)

    i += 1
